ORStools/proc/base_processing_algorithm.py

```diff
@@ -151,14 +188,41 @@ def _get_ors_client_from_provider(provider: str, feedback: QgsProcessingFeedback
         ors_client.overQueryLimit.connect(lambda: feedback.reportError("OverQueryLimit: Retrying..."))
         return ors_client
 
+
+    def parseOptions(self, parameters: dict, context: QgsProcessingContext) -> dict:
+        options = dict()
+
+        if parameters[self.IN_AVOID_FEATS]:
+            options['avoid_features'] = [dict(enumerate(AVOID_FEATURES))[feat] for feat in parameters[self.IN_AVOID_FEATS]]
+
+        if parameters[self.IN_AVOID_BORDERS]:
+            options['avoid_features'] = [dict(enumerate(AVOID_BORDERS))[feat] for feat in parameters[self.IN_AVOID_BORDERS]]
+
+        countries_raw = parameters[self.IN_AVOID_COUNTRIES]
+        countries = None
+        if countries_raw:
+            countries = list(map(int, countries_raw.split(',')))
+            options['avoid_countries'] = countries
+
+        polygons_layer = self.parameterAsLayer(parameters, self.IN_AVOID_POLYGONS, context)
+        polygons = None
+        if polygons_layer:
+            polygons = _get_avoid_polygons(polygons_layer)
+            options['avoid_polygons'] = polygons
+
+        return options
+
     # noinspection PyUnusedLocal
     def initAlgorithm(self, configuration):
         """
         Combines default and algorithm parameters and adds them in order to the
         algorithm dialog window.
         """
-        parameters = [self.provider_parameter()] + self.PARAMETERS + [self.output_parameter()]
+        parameters = [self.provider_parameter(), self.profile_parameter()] + self.PARAMETERS + self.option_parameters() + [self.output_parameter()]
         for param in parameters:
+            if param.name() in ADVANCED_PARAMETERS:
+                param.setFlags(param.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
```

Comment:
What does the `(param.flags() | QgsProcessingParameterDefinition.FlagAdvanced)` do?
So the Advanced flag can be overwrtitten by a flag specified for a single parameter?

Reply:
No, the `|` here is the union operator for sets.

Reply:
i see 👍🏻, thought it was like an `or` in JavaScript (`||`)

Reply:
It… is, in a way.
The backing type is still `int`, so this can also be seen as a bitwise `or` (which is what might happen in the background, not too sure…).

Reply:
hmm, we could use `.union()` instead to make the relation explicit. Either is fine.

Reply:
No, `flags()` has no `union()`-attribute. I think it's actually a bitwise `or`, made type-safe by QFlags. A comment might be better suited to explain what's going on. I'll add one :)

Reply:
damn, thoght its a set :), comment sounds good

Reply:
done